File: tests/browser/features/support/pages/non_existing_lexeme_page.rb
class NonExistingLexemePage
  include PageObject
  page_url ENV['WIKIDATA_REPO_URL'] + ENV['LEXEME_NAMESPACE'] + 'L-invalid'

  h1(:first_heading, class: 'firstHeading')
  div(:no_article_text, class: 'noarticletext')
end

File: cypress/support/pageObjects/NonExistingLexemePage.ts
export class NonExistingLexemePage {

	private static get NON_EXISTING_LEXEME_PAGE_SELECTORS(): Record<string, string> {
		return {
			FIRST_HEADING: '#firstHeading',
			NO_ARTICLE_TEXT: '.noarticletext'
		};
	}

	public firstHeading(): Chainable {
		return cy.get( this.constructor.NON_EXISTING_LEXEME_PAGE_SELECTORS.FIRST_HEADING );
	}

	public noArticleText(): Chainable {
		return cy.get( this.constructor.NON_EXISTING_LEXEME_PAGE_SELECTORS.NO_ARTICLE_TEXT );
	}

	public open(): this {
		cy.visitTitle( { title: 'Lexeme:L-invalid', failOnStatusCode: false } );
		return this;
	}
}
